dm: add a note about DM to `sql_require_primary_key` (#15993)
pingcap/docs: system-variables.md

- This variable controls whether to enforce the requirement that a table has a primary key. After this variable is enabled, attempting to create or alter a table without a primary key will produce an error.
- This feature is based on the similarly named [`sql_require_primary_key`](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_sql_require_primary_key) in MySQL 8.0.
- It is strongly recommended to enable this variable when using TiCDC. This is because replicating changes to a MySQL sink requires that tables have a primary key.

> - When enabling this variable and using TiDB Data Migration (DM) for data migration, set it to 'OFF' in the 'session' parameter of the DM task configuration to avoid DM task creation failure.

Reviewer: ```suggestion
- If you enable this variable and are using TiDB Data Migration (DM) to migrate data, it is recommended that you set `sql_require_ primary_key` to `OFF` in the `session` part in the [DM Task Configuration File](/dm/task-configuration-file-full.md#task-configuration-file-template-advanced). Otherwise it will cause the DM fail to create tasks.
```

- This feature is based on the similarly named [`sql_require_primary_key`](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_sql_require_primary_key) in MySQL 8.0.
- It is strongly recommended to enable this variable when using TiCDC. This is because replicating changes to a MySQL sink requires that tables have a primary key.

<CustomContent platform="tidb">

- If you enable this variable and are using TiDB Data Migration (DM) to migrate data, it is recommended that you add `sql_require_ primary_key` to the `session` part in the [DM Task Configuration File](/dm/task-configuration-file-full.md#task-configuration-file-template-advanced) and set it to `OFF`. Otherwise, it will cause DM to fail to create tasks.

</CustomContent>

> 
> <CustomContent platform="tidb-cloud">
> 
> </CustomContent>

Reviewer 2: ```suggestion
```

- This feature is based on the similarly named [`sql_require_primary_key`](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_sql_require_primary_key) in MySQL 8.0.
- It is strongly recommended to enable this variable when using TiCDC. This is because replicating changes to a MySQL sink requires that tables have a primary key.

<CustomContent platform="tidb">

> - If you enable this variable and are using TiDB Data Migration (DM) to migrate data, it is recommended that you set `sql_require_ primary_key` to `OFF` in the `session` part in the [DM Task Configuration File](/dm/task-configuration-file-full.md#task-configuration-file-template-advanced). Otherwise it will cause the DM fail to create tasks.

Reviewer 2: ```suggestion
- If you enable this variable and are using TiDB Data Migration (DM) to migrate data, it is recommended that you add `sql_require_ primary_key` to the `session` part in the [DM Task Configuration File](/dm/task-configuration-file-full.md#task-configuration-file-template-advanced) and set it to `OFF`. Otherwise, it will cause DM to fail to create tasks.
```

</CustomContent>

<CustomContent platform="tidb-cloud">

> - If you enable this variable and are using TiDB Data Migration (DM) to migrate data, it is recommended that you set `sql_require_ primary_key` to `OFF` in the `session` level in the [DM Task Configuration File](https://docs.pingcap.com/tidb/stable/task-configuration-file-full#task-configuration-file-template-advanced). Otherwise it will cause DM to fail to create tasks.

Author: TiDB Cloud 没有开放配置文件给用户修改，所以云上用户是无法修改该参数的。不过视乎云上也没提供修改该系统变量的地方https://docs.pingcap.com/tidbcloud/system-variables#sql_require_primary_key-span-classversion-marknew-in-v630span

Reviewer 2: @Frank945946  而且云上的 TiDB Data Migration (DM) 是指界面上的功能，那云上就不显示这条？

Reviewer: 删掉了

> - If you enable this variable and are using TiDB Data Migration (DM) to migrate data, it is recommended that you set `sql_require_ primary_key` to `OFF` in the `session` level in the [DM Task Configuration File](https://docs.pingcap.com/tidb/stable/task-configuration-file-full#task-configuration-file-template-advanced). Otherwise it will cause DM to fail to create tasks.

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```
